metrics/internal/collectors/operator-condition.go

```diff
@@ -0,0 +1,192 @@
+			`Operator Upgradeable Status; 0: Upgradeable (True), 1: Not Upgradeable (False), 2: Unknown`,
+			[]string{"name", "namespace", "reason", "message"},
+			nil,
+		),
+		Informer:             sharedIndexInformer,
+		operatorNamespace:    operatorNamespace,
+		operatorNamePrefixes: getDefaultOperatorPrefixes(),
+	}
+}
+
+func (c *OperatorConditionCollector) Run(stopCh <-chan struct{}) {
+	go c.Informer.Run(stopCh)
+}
+
+func (c *OperatorConditionCollector) Describe(ch chan<- *prometheus.Desc) {
+	ds := []*prometheus.Desc{
+		c.UpgradeableStatus,
+	}
+	for _, d := range ds {
+		ch <- d
+	}
+}
+
+func (c *OperatorConditionCollector) Collect(ch chan<- prometheus.Metric) {
+	ocLister := NewOperatorConditionLister(c.Informer.GetIndexer())
+	operatorConditions := c.getFilteredOperatorConditions(ocLister)
+	if len(operatorConditions) == 0 {
+		return
+	}
+	c.collectUpgradeableStatus(ch, operatorConditions)
+}
+
+// getFilteredOperatorConditions retrieves and filters operator conditions based on configured prefixes
+func (c *OperatorConditionCollector) getFilteredOperatorConditions(lister Lister[operatorv2.OperatorCondition]) []*operatorv2.OperatorCondition {
+	operatorConditions, err := lister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("couldn't list OperatorConditions: %v", err)
+		return nil
+	}
+
+	return c.filterOperatorConditions(operatorConditions)
+}
+
+// filterOperatorConditions filters operator conditions based on configured name prefixes
+func (c *OperatorConditionCollector) filterOperatorConditions(conditions []*operatorv2.OperatorCondition) []*operatorv2.OperatorCondition {
+	var filtered []*operatorv2.OperatorCondition
+	for _, oc := range conditions {
+		if c.matchesOperatorPrefix(oc.Name) {
+			filtered = append(filtered, oc)
+		}
+	}
+	return filtered
+}
+
+// matchesOperatorPrefix checks if the given name matches any configured operator prefix
+func (c *OperatorConditionCollector) matchesOperatorPrefix(name string) bool {
+	for _, prefix := range c.operatorNamePrefixes {
+		if strings.HasPrefix(name, prefix) {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *OperatorConditionCollector) collectUpgradeableStatus(ch chan<- prometheus.Metric, operatorConditions []*operatorv2.OperatorCondition) {
+	for _, operatorCondition := range operatorConditions {
+		status := 2 // Unknown by default
+		reason := "Unknown"
+		message := "Upgradeable condition not found"
+
+		// Check spec.overrides first (highest priority)
+		if len(operatorCondition.Spec.Overrides) > 0 {
+			if s, r, m, found := findUpgradeableCondition(operatorCondition.Spec.Overrides); found {
+				status = s
+				reason = r
+				message = m
+				// If upgradeable is overridden, skip checking spec.conditions
+				ch <- prometheus.MustNewConstMetric(
+					c.UpgradeableStatus,
+					prometheus.GaugeValue,
+					float64(status),
+					operatorCondition.Name,
+					operatorCondition.Namespace,
+					reason,
+					message,
```

Comment:
to avoid reason and message being empty strings, we should provide some sort of a default empty string like, reason is "None" and message is "NotSpecified"